pkg/config/config_template.yaml

```diff
@@ -1,4 +1,4 @@
-{{ if .Common }}
+{{- if .Common -}}
```

Comment:
@hush-hush 
> Have we tried uncommenting all the lines in the new template and check that the Agent can still start ?

It begins to start but fails in the end. I think that means it parsed the YAML without errors at least? I think there are just a lot of invalid configs, for example
```yaml
proxy:
  https: http://<USERNAME>:<PASSWORD>@<PROXY_SERVER_FOR_HTTPS>:<PORT>
  http: http://<USERNAME>:<PASSWORD>@<PROXY_SERVER_FOR_HTTP>:<PORT>
log_file: <AGENT_LOG_FILE_PATH>
```

in my case it crashes from a null-deref in the SNMP check due to a missing err check.
https://github.com/DataDog/datadog-agent/blob/35654a389911966ebe8430bf8338e18e67ad3ea1/pkg/snmp/devicededuper/device_deduper.go#L71-L72